### PR TITLE
Test/front end unit tests

### PR DESCRIPTION
**Pull request by lposo**
Added tests for the Graph, GraphView, DataCell and NavigationMenu. For most tests, in order to verify if a function was called, a jest spy would be linked to the console log. This spy would check to see if certain messages were called in order to validate the functions. Other tests added include snapshot tests to see if anything changed in regards to the page.

### CODE REVIEW

**Review by Mehrdadmaskull — COMMENTED** [review]
@lposo LGTM, but do we need the snapshot files too?

**Comment by lposo** [comment]
@Mehrdadmaskull  They're auto generated when you run the snapshot tests. I guess we can remove them though.

**Review by Mehrdadmaskull — APPROVED** [review]
LGTM

**Comment by lposo** [comment]
Currently, the tests simply verify if components were rendered, then simulate clicking buttons in order to see if the corresponding methods were called.

File: Client/src/Components/Graph/Graph.js [hunk 1]
```diff
@@ -22,10 +22,12 @@ export default function Graph(props) {
 
   const handleXScaleClick = () => {
     setXToggle(!isXLog)
+    console.log('Changed X: ' + isXLog)
```
Comment (TiraFire) [inline Client/src/Components/Graph/Graph.js:25]: This is a good way to test the change in a function that returns nothing.

File: Client/src/Components/Graph/Graph.js [hunk 2]
```diff
@@ -226,8 +228,8 @@ export default function Graph(props) {
         preserveAspectRatio="xMidYMid meet">
       </svg>
       <div>
-        <Button onClick={handleXScaleClick} color="primary">Change X Scale</Button>
-        <Button onClick={handleYScaleClick} color="primary">Change Y Scale</Button>
+        <Button id='btn1' onClick={handleXScaleClick} color="primary">Change X Scale</Button>
+        <Button id='btn2' onClick={handleYScaleClick} color="primary">Change Y Scale</Button>
       </div>
```
Comment (TiraFire) [inline Client/src/Components/Graph/Graph.js:233]: Ids to buttons should be standard, thank you for adding it

**Review by TiraFire — APPROVED** [review]
Looks good, I am happy that there could be some unit tests for the small functions in the front end even if it is just testing the call.

**Pull request merged**